.github/actions/dump-ci-stats-to-gcp-metrics.cjs

```diff
@@ -0,0 +1,155 @@
+  };
+  try {
+    await monitoring.createTimeSeries(request);
+    console.log(`Metric ${metricType} sent successfully.`);
+  } catch (error) {
+    console.error('Error sending metric:', error);
+  }
+}
+
+// Function to fetch workflow and job details via GitHub API
+async function fetchWorkflowDetails() {
+  const runId = process.argv[2];   
+  const repo = process.env.GITHUB_REPOSITORY;
+  const apiUrl = `https://api.github.com/repos/${repo}/actions/runs/${runId}`;
+
+  try {
+    const response = await fetch(apiUrl, {
+      headers: {
+        'Authorization': `Bearer ${process.env.GITHUB_TOKEN}`,
+        'Accept': 'application/vnd.github.v3+json'
+      }
+    });
+
+    if (!response.ok) throw new Error(`HTTP error! status: ${response.status}`);
+    const data = await response.json();
+
+    return {
+      workflowId: data.id,
+      workflowName: data.name,
+      status: data.status, // "completed", "in_progress", etc.
+      conclusion: data.conclusion, // "success", "failure"
+      startTime: data.created_at,
+      endTime: data.updated_at,
+      trigger: data.event, // "push", "pull_request", etc.
+      jobs: await fetchJobDetails(repo, data.id) // Fetch individual job details
+    };
+  } catch (error) {
+    console.error('Error fetching workflow details:', error);
+    process.exit(1);
+  }
+}
+
+async function fetchJobDetails(repo, runId) {
+  const apiUrl = `https://api.github.com/repos/${repo}/actions/runs/${runId}/jobs`;
+
+  try {
+    const response = await fetch(apiUrl, {
+      headers: {
+        'Authorization': `Bearer ${process.env.GITHUB_TOKEN}`,
+        'Accept': 'application/vnd.github.v3+json'
+      }
+    });
+
+    if (!response.ok) throw new Error(`HTTP error! status: ${response.status}`);
+    const data = await response.json();
+    return data.jobs;
+  } catch (error) {
+    console.error('Error fetching job details:', error);
+    return [];
+  }
+}
+
+// Main function to send metrics
+(async () => {
+  try {
+    const workflowStats = await fetchWorkflowDetails();
+
+    // Send workflow-level metrics
+    const workflowLabels = {
+      workflow_name: workflowStats.workflowName,
+      workflow_id: workflowStats.workflowId,
+      trigger: workflowStats.trigger
+    };
+
+    const workflowDuration = (new Date(workflowStats.endTime) - new Date(workflowStats.startTime)) / 1000;
+    await sendMetricsToGCP('ci_workflow_duration', workflowDuration, workflowLabels);
+
+    // Send job-level metrics for each job in the workflow
+    for (const job of workflowStats.jobs) {
+      const jobLabels = {
+        workflow_name: workflowStats.workflowName,
+        job_name: job.name,
+        runner_name: job.runner_name,
+        conclusion: job.conclusion
+      };
+
+      const jobExecutionTime = (new Date(job.completed_at) - new Date(job.started_at)) / 1000;
+      await sendMetricsToGCP('ci_job_execution_time', jobExecutionTime, jobLabels);
+
+      // Send job status (1 for success, 0 for failure)
```

Comment:
Will see if we can have a separate number for a cancelled job. It is 0 for cancel too